source/Planet.h

```diff
@@ -37,6 +37,39 @@ class System;
 // are available, as well as attributes that determine what sort of missions
 // might choose it as a source or destination.
 class Planet {
+public:
+	class Port
+	{
+	public:
+		// The different properties that can be recharged by a port.
+		enum RechargeType
```

Comment:
As mentioned in my comment, on top of a RechargeType could we have a ServicesType?
```
enum ServicesType
{
	None = 0,
	Trading = (1 << 0),
	JobBoard = (1 << 1),
	Bank = (1 << 2),
	HireExtraCrew = (1 << 3),
	OffersMissions = (1 << 4),
	All = Trading | JobBoard | Bank | HireExtraCrew | OffersMissions,
};
```
I don't see customizable spaceport services as too much of a leap from customizable port recharge types.
A default spaceport would then just use ServicesType::All.